snapcraft/plugins/go.py

```diff
@@ -101,7 +102,13 @@ def get_pull_properties(cls):
 
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
-        self.build_packages.append('golang-go')
+
+        # Do not try to install golang if it is already available in our PATH.
+        try:
+            subprocess.check_output(['go', 'version'])
+        except OSError:
+            self.build_packages.append('golang-go')
```

Comment:
My preferred solution would be *not* to install any packages here.

Odds are somebody building a snap from golang sources already has the appropriate golang toolchain set-up.

Preferred solution:
raise OSError("A golang compiler cannot be found. Please make sure an appropriate compiler can be found in your PATH.")